spec.mediawiki

```diff
@@ -407,50 +375,149 @@ Decred blockchain may be useful for establishing a reputation system.
-trusted nodes for each of the assets supported.
+'''Clients''' need atomic-swap enabled wallets and the ability to broadcast 
+transactions to each of the blockchain networks involved in 
+[[#Atomic_Settlement|the swap]].
+
+'''DEX operators''' need access to trusted full nodes for each of the 
+assets supported.
 While operation via a surrogate blockchain data service such as a block explorer 
 is potentially feasible, it would entail significant security risks.
-Initial development will use only a direct connection to full nodes of each 
-supported asset's blockchain.
-
-With the exception of Decred and Bitcoin, which will be implemented during 
-initial phases of DEX development, project developers will be responsible for 
-adding support for their particular asset. 
+Initial development will require a direct connection to full nodes of each 
+asset's blockchain.
 
 Adding support for an asset is accomplished by implementing a particular 
 [https://golang.org/ Go] interface. 
 The interface is exported and compiled with `-buildmode=plugin` to enable 
 dynamic importing by the exchange operator. 
-The operator will decide which plugins to import in the DEX configuration. 
+The operator will decide which plugins to import via configuration file.
+As the expected interface is expected to evolve greatly during initial
+implementation, the specifics of interface structure will be released at a later
+time.
 
-<pre>
-type DExAsset interface {
-  // Initialize the asset. An channel is passed that will receive a signal
-  // when a new block is added to the chain. 
-  // The path to a configuration file is the only other argument. 
-  // The configuration file format is not formalized, so the asset developer
-  // will need to publish instructions for configuration.
-  Initialize(chan struct{}, string) error
-  // UTXO retrieves info about the transaction output.
-  UTXO(string, int) (DExUTxO, error)
-  // CheckSig validates that the signature for the provided data is a valid 
-  // signature associated with the public key.
-  CheckSig(txid string, vout int, pubkey, data, sig, extradata []byte) (bool, error)
-}
+With the exception of Decred and Bitcoin, which will be implemented during 
+initial phases of DEX development, it is expected that development communities 
+will release their own appropriately vetted plugins. 
 
-type DExUTxO interface {
-	Confirmations() int
-	Valid() bool
-	Address() string
-	Lock() [32]byte
-	Key() [32]byte
-}
-</pre>
+==Account Creation==
+
+An account is uniquely identified by the client's public account key (pubkey), 
+which the client provides during registration.
+After registration, the client does identify themselves with their pubkey 
+directly. 
+Instead, the account is identified by an '''account ID''', which is the first 16
+bytes of the double SHA-256 hash of the client's pubkey,
+'''''<code>sha256(sha256(pubkey))[:16]</code>''''', provided as a hex-encoded 
```

Comment:
The intention was space savings. The order ID is the longest member of the Order object and is sent at least once to every client subscribed to the market's order book. A halving of the length is a significant reduction to bandwidth. 

I can't say much about the security implications except that 256^16 = 3.4e38 so collisions are probably unlikely. 

During editing, I had the JSON encoding written as hex, base58, and base64 at various times. I think the actual savings from base64 is something like 35% of the length, so substantial. I guess we just need to be clear that the serialization is based on integer equivalent interpretation into a length 16 byte array, rather than an ASCII encoding, which could have variable length.

Reply:
I'm just concerned we're optimizing where it's not needed.  I suspect the Json payloads will be less than typical MTUs regardless of the chosen encoding.  Maybe an opinion from @jrick?

Reply:
I think you're probably right. Then hex or base64 JSON encoding?

Reply:
Hex or base64 work fine.  I'd probably take hex for shorter values (base64 is more space efficient but this only matters for larger values, see https://play.golang.org/p/5dGua6v9cwU)

Reply:
@jrick How about the idea of taking the first 16 bytes of the hash rather than the entire hash.  Risky in terms of security, or totally fine?

Reply:
that depends on how it's used (which I'm not familiar with, haven't read the spec)

Reply:
I've set it to full 32 for now. We can shorten it later time if desired.